integration-tests/src/test/java/tachyon/master/next/JournalShutdownIntegrationTest.java

```diff
@@ -0,0 +1,203 @@
+
+/**
+ * Test master journal for cluster terminating. Assert that test can replay the editlog and
+ * reproduce the correct state. Test both the single master(tachyon) and multi masters(tachyon-ft).
+ */
+public class JournalShutdownIntegrationTest {
+
+  /**
+   * Hold a client and keep creating files/tables.
+   */
+  class ClientThread implements Runnable {
+    /** The number of successfully created files/tables. */
+    private int mSuccessNum = 0;
+
+    private final int mOpType; // 0:create file; 1:create raw table.
+    private final TachyonFS mTfs;
+
+    public ClientThread(int opType, TachyonFS tfs) {
+      mOpType = opType;
+      mTfs = tfs;
+    }
+
+    public int getSuccessNum() {
+      return mSuccessNum;
+    }
+
+    /**
+     * Keep creating files/tables until something crashes or fail to create. Record how many files/
+     * tables are created successfully.
+     */
+    @Override
+    public void run() {
+      try {
+        // This infinity loop will be broken if something crashes or fail to create. This is
+        // expected since the master will shutdown at a certain time.
+        while (true) {
+          if (mOpType == 0) {
+            try {
+              mTfs.getOutStream(new TachyonURI(TEST_FILE_DIR + mSuccessNum),
+                  ClientOptions.defaults()).close();
+            } catch (IOException ioe) {
+              break;
+            }
+          } else if (mOpType == 1) {
+            // TODO Add this back when there is new RawTable client API
+            // if (mTfs.createRawTable(new TachyonURI(TEST_TABLE_DIR + mSuccessNum), 1) == -1) {
+            // break;
+            // }
+          }
+          mSuccessNum ++;
+          CommonUtils.sleepMs(null, 100);
+        }
+      } catch (Exception e) {
+        // Something crashed. Stop the thread.
+      }
+    }
+  }
+
+  private static final int TEST_BLOCK_SIZE = 128;
+  private static final String TEST_FILE_DIR = "/files/";
+  private static final int TEST_NUM_MASTERS = 3;
+  private static final String TEST_TABLE_DIR = "/tables/";
+  private static final long TEST_TIME_MS = Constants.SECOND_MS;
+
+  private ClientThread mCreateFileThread = null;
+  private ClientThread mCreateTableThread = null;
+  /** Executor for running client threads */
+  private final ExecutorService mExecutorsForClient = Executors.newFixedThreadPool(2);
+  /** Executor for constructing MasterInfo */
+  private final ExecutorService mExecutorsForMasterInfo = Executors.newFixedThreadPool(2);
+  private TachyonConf mMasterTachyonConf = null;
+
+  @After
+  public final void after() throws Exception {
+    mExecutorsForClient.shutdown();
+    mExecutorsForMasterInfo.shutdown();
+    System.clearProperty("fs.hdfs.impl.disable.cache");
+  }
+
+  @Before
+  public final void before() throws Exception {
+    System.setProperty("fs.hdfs.impl.disable.cache", "true");
+  }
+
+  private FileSystemMaster createFsMasterFromJournal() throws IOException {
+    // TODO
```

Comment:
This method should be the same with [this method in another PR](https://github.com/amplab/tachyon/pull/1443/files#diff-5deda5f174aef4333820ae72681496afR84), would update after that PR is merged.